### PR TITLE
Added validation split, compute validation loss in training

### PR DESCRIPTION
**Pull request by dribnet**
Quick attempt to add a validation split and compute the validation loss when training. The general idea is to try to get a sense if the model starts to overfit to the data it has seen. This also provides a more deterministic metric is choosing hyperparamaters across runs.

This could probably be cleaned up and I'm not sure if this follows tensorflow idioms, etc. But it seems to be working in the current form. I'm curious to get feedback if others think this change could be a useful metric during training.

### CODE REVIEW

**Comment by hardmaru** [comment]
Thx!
